library/Icingadb/Widget/HostListItemDetailed.php

```diff
@@ -32,8 +33,17 @@ protected function assembleFooter(BaseHtmlElement $footer)
         $statusIcons = new HtmlElement('div', Attributes::create(['class' => 'status-icons']));
 
         // ToDo(fs): Get `has_comments` from database
-        if ($this->item->comment->limit(1)->execute()->hasResult()) {
-            $statusIcons->addHtml(new Icon('comments', ['title' => t('This item has been commented')]));
+        if (! empty($this->item->state->last_comment->text)) {
+            $comment = $this->item->state->last_comment;
+            $comment->host = $this->item;
+            $comment = (new CommentList([$comment]))
+                ->setNoSubjectLink()
+                ->setDetailActionsDisabled();
+
+            $statusIcons->addHtml(
+                (new Icon('comments', ['class' => 'popup']))
+                ->add(new HtmlElement('div', Attributes::create(['class' => 'comment-popup']), $comment))
```

Comment:
You're adding the popup (block element) to the icon (inline element). That's not valid HTML.